setup.py

```diff
@@ -13,29 +10,36 @@
 from setuptools.command.build_ext import build_ext
 from shutil import copyfile, copymode
 
+
 class CMakeExtension(Extension):
     def __init__(self, name, sourcedir=''):
         Extension.__init__(self, name, sources=[])
         self.sourcedir = os.path.abspath(sourcedir)
 
+
 class CMakeBuild(build_ext):
 
     build_ext.user_options = build_ext.user_options + [
+        # Notes: the first option is the option string
+        #        the second option is an abbreviated form of an option, which we avoid with None
         ('ambitpath', None, 'the path to ambit'),
-        ('max_det_orb', 64, 'the maximum number of orbitals used by the Determinant class'),
-        ('enable_codecov', False, 'enable code coverage'),
-        ('cmake_config_options', '', 'cmake configuration'),
-        ('cmake_build_options', '', 'cmake build options')
-        ]
+        ('max-det-orb', None, 'the maximum number of orbitals used by the Determinant class'),
+        ('enable-codecov', None, 'enable code coverage'),
+        ('cmake-config-options', None, 'cmake configuration'),
+        ('cmake-build-options', None, 'cmake build options')
+    ]
 
     def initialize_options(self):
         self.ambitpath = None
         self.max_det_orb = 64
         self.enable_codecov = 'OFF'
         self.cmake_config_options = ''
-        ncore_phys = psutil.cpu_count(logical=False)
-        ncore_virt = psutil.cpu_count(logical=True)
-        nprocs = ncore_phys + 1 if ncore_phys != ncore_virt else ncore_phys
+        # get the CPU count. None is returned if the number of CPUs is undeterminded.
+        cpu_count = os.cpu_count()
+        if cpu_count is None:
+            nprocs = 1
+        # choose a reasonable number of processes
+        nprocs = cpu_count / 2 + 1 if cpu_count > 4 else cpu_count
```

Comment:
We may want to allow the user to pass this as a command line argument and use this logic if the argument isn't passed. This may well be outside the scope of the PR.

Reply:
It is possible. The `cmake-build-options` should override this. For example, `make -j2 -j8` reads the last one.

Reply:
Here I decided that we should go with the maximum number and let the use set a particular value if needed. I added the variable nprocs that can be specified in the setup.cfg file.